src/utils_request.js

```diff
@@ -0,0 +1,272 @@
+        switch (compression) {
+        case 'gzip':
+            decompressor = zlib.createGunzip();
+            break;
+        case 'deflate':
+            decompressor = zlib.createInflate();
+            break;
+        case 'br':
+            decompressor = decompressStream();
+            break;
+        case undefined:
+            break;
+        default:
+            throw new Error(`requestBetter: Invalid Content-Encoding header. Expected gzip, deflate or br, but received: ${compression}`);
+        }
+        stream = response.pipe(decompressor);
+    }
+
+    return new Promise((resolve, reject) => {
+        const chunks = [];
+        stream
+            .on('data', chunk => chunks.push(chunk))
+            .on('error', err => reject(err))
+            .on('end', () => {
+                const buffer = Buffer.concat(chunks);
+                resolve(buffer.toString(encoding));
+            });
+    });
+}
+/**
+ * Sends a HTTP request that looks like a request sent by a web browser,
+ * fully emulating browser's HTTP headers.
+ *
+ * This function is useful for web scraping of websites that send the full HTML in the first response.
+ * Thanks to this function, the target web server has no simple way to find out the request
+ * hasn't been sent by a full web browser. Using a headless browser for such requests
+ * is an order of magnitude more resource-intensive than this function.
+ *
+ * Currently, the function sends requests the same way as Firefox web browser does.
+ * In the future, it might add support for other browsers too.
+ *
+ * Internally, the function uses `requestBetter()` function to perform the request.
+ * All `options` not recognized by this function are passed to it,
+ * so see it for more details.
+ *
+ * @param options.url
+ *  URL of the target endpoint. Supports both HTTP and HTTPS schemes.
+ * @param [options.method=GET]
+ *  HTTP method.
+ * @param [options.headers={}]
+ *  Additional HTTP headers to add. It's recommended not to use this option,
+ *  because it can ruin the signature of the web browser. TODO: Maybe let's remove this completely?
+ * @param [options.languageCode=en]
+ *  Two-letter ISO 639 language code.
+ * @param [options.countryCode=US]
+ *  Two-letter ISO 3166 country code.
+ * @param [options.isMobile]
+ *  If `true`, the function uses User-Agent of a mobile browser.
+ *
+ * @return {{ response, body }}
+ *  Returns an object with two properties: `response` is the instance of
+ *  Node's [`http.IncomingMessage`](https://nodejs.org/api/http.html#http_class_http_incomingmessage) class,
+ *  `body` is a `String`, `Buffer` or `Object`, depending on the `encoding` and `json` options.
+ */
+export const requestLikeBrowser = async (options) => {
+    const defaultOptions = {
+        countryCode: 'US',
+        languageCode: 'en',
+        headers: {},
+        method: 'GET',
+        useMobileVersion: false,
+    };
+    const opts = Object.assign(defaultOptions, options);
```

Comment:
` Object.assign({one: 1}, {one: undefined})` produces `{one: undefined}`. Not sure that we want this.